src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ossirian.cpp

```diff
@@ -94,9 +98,33 @@ struct boss_ossirian : public BossAI
     void Reset() override
     {
         BossAI::Reset();
+
+        Creature* trigger = me->GetMap()->GetCreature(_triggerGUID[0]);
+        if (trigger)
+        {
+            trigger->DespawnOrUnsummon();
+            if (GameObject* crystal = me->GetMap()->GetGameObject(_crystalGUID[0]))
+                crystal->Delete();
+        }
+
+        if (trigger = me->GetMap()->SummonCreature(NPC_OSSIRIAN_TRIGGER, initialCrystalPosition))
```

Comment:
![image](https://user-images.githubusercontent.com/16887899/183309512-0a9bc6c6-f851-4b8c-8d04-30e50a921251.png)
```suggestion
        if (trigger == me->GetMap()->SummonCreature(NPC_OSSIRIAN_TRIGGER, initialCrystalPosition))
```

Reply:
No. It's not comparison, but assigment.